src/overlays/actors/ovl_En_Ssh/z_en_ssh.c

```diff
@@ -21,95 +63,798 @@ const ActorInit En_Ssh_InitVars = {
+        radius *= scale * radiusMod;
+        height *= scale;
+
+        this->colCylinder[i].dim.yShift = yShift;
+        this->colCylinder[i].dim.radius = radius;
+        this->colCylinder[i].dim.height = height;
+    }
+    Actor_SetScale(&this->actor, 0.04f * scale);
+    this->groundYoffset = 40.0f * scale;
+    this->colliderScale = scale * 1.5f;
+}
+
+s32 EnSsh_Damaged(EnSsh* this) {
+    if ((this->stunTimer == 120) && (this->stateFlags & STATE_STUNNED)) {
+        func_8003426C(&this->actor, 0, 0xC8, 0, this->stunTimer);
+    }
+    if (DECR(this->stunTimer) != 0) {
+        Math_SmoothScaleMaxMinS(&this->maxTurnRate, 0x2710, 0xA, 0x3E8, 1);
+        return false;
+    } else {
+        this->stunTimer = 0;
+        this->stateFlags &= ~STATE_STUNNED;
+        this->spinTimer = 0;
+        if (this->swayTimer == 0) {
+            this->spinTimer = 30;
+        }
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALTU_ROLL);
+        Audio_PlayActorSound2(&this->actor, NA_SE_VO_ST_ATTACK);
+        return true;
+    }
+}
+
+void EnSsh_Turn(EnSsh* this, GlobalContext* globalCtx) {
+    if (this->hitTimer != 0) {
+        this->hitTimer--;
+    }
+    if (DECR(this->spinTimer) != 0) {
+        this->actor.posRot.rot.y += 10000.0f * (this->spinTimer / 30.0f);
+    } else if ((this->swayTimer == 0) && (this->stunTimer == 0)) {
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 4, 0x2710, 1);
+    }
+    this->actor.shape.rot.y = this->actor.posRot.rot.y;
+}
+
+void EnSsh_Stunned(EnSsh* this, GlobalContext* globalCtx) {
+    if ((this->swayTimer == 0) && (this->stunTimer == 0)) {
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink ^ 0x8000, 4, this->maxTurnRate, 1);
+    }
+    this->actor.shape.rot.y = this->actor.posRot.rot.y;
+    if (this->stunTimer < 30) {
+        if (this->stunTimer & 1) {
+            this->actor.shape.rot.y += 0x7D0;
+        } else {
+            this->actor.shape.rot.y -= 0x7D0;
+        }
+    }
+}
+
+void EnSsh_UpdateYaw(EnSsh* this, GlobalContext* globalCtx) {
+    if (this->stunTimer != 0) {
+        EnSsh_Stunned(this, globalCtx);
+    } else {
+        EnSsh_Turn(this, globalCtx);
+    }
+}
+
+void EnSsh_Bob(EnSsh* this, GlobalContext* globalCtx) {
+    f32 bobVel = 0.5f;
+
+    if ((globalCtx->state.frames & 8) != 0) {
+        bobVel *= -1.0f;
+    }
+    Math_SmoothScaleMaxMinF(&this->actor.velocity.y, bobVel, 0.4f, 1000.0f, 0.0f);
+}
+
+s32 EnSsh_IsCloseToLink(EnSsh* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 yDist;
+
+    if (this->stateFlags & STATE_GROUND_START) {
+        return true;
+    }
+    if (this->unkTimer != 0) {
+        return true;
+    }
+    if (this->swayTimer != 0) {
+        return true;
+    }
+    if (this->animTimer != 0) {
+        return true;
+    }
+
+    if (this->actor.xzDistFromLink > 160.0f) {
+        // player is more than 160 xz units from the skulltulla
```

Comment:
These comments are unnecessary imo, since they don't add anything and the conditions are already very easy to understand.